### ISSUE TITLE
Memory leak on 1.20.14

### ISSUE THREAD

**Issue by xcompass** [kubernetes/kubernetes#108908]
### What happened?

Please see details here: https://stackoverflow.com/questions/71519202/kubernetes-memory-leak-troubleshooting

### What did you expect to happen?

No memory leak

### How can we reproduce it (as minimally and precisely as possible)?

I think the memory leak is there but amplified by cronjob, which is frequently creating and tear down the pods. So to reproduce the issue, one would setup a few cronjobs to run every minute on a node running 1.20.14.

### Anything else we need to know?

I provisioned another node with 1.19.16. There is no memory leak on that node. I also noticed that on 1.19.16 node, under /sys/fs/cgroup, there is not kubepods directory, only kubepods.slice, where on 1.20.14 node, both exist. And all those empty pods in the screenshots are under kubepods directory. I wonder if those pods are placed in wrong directory and didn't get cleaned up. 

Happy to provide more information if needed

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-25T21:17:57Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"darwin/arm64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.14", GitCommit:"57a3aa3f13699cf3db9c52d228c18db94fa81876", GitTreeState:"clean", BuildDate:"2021-12-15T14:47:10Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>
On prem, VMware/vCloud
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
NAME="Flatcar Container Linux by Kinvolk"
ID=flatcar
ID_LIKE=coreos
VERSION=3033.2.3
VERSION_ID=3033.2.3
BUILD_ID=2022-03-02-2032
PRETTY_NAME="Flatcar Container Linux by Kinvolk 3033.2.3 (Oklo)"
ANSI_COLOR="38;5;75"
HOME_URL="https://flatcar-linux.org/"
BUG_REPORT_URL="https://issues.flatcar-linux.org"
FLATCAR_BOARD="amd64-usr"
$ uname -a
Linux f6.workers.ctlt.ubc.ca 5.10.102-flatcar #1 SMP Wed Mar 2 19:40:13 -00 2022 x86_64 Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz GenuineIntel GNU/Linux

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

- RuntimeName:  containerd
- RuntimeVersion:  1.5.9
- RuntimeApiVersion:  v1alpha2

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by SergeyKanzhelev** [comment]
https://github.com/kubernetes/kubernetes/issues/100241 this may have fixed it but wasn't backported to 1.20. Please update to a supported version and try again

/close

**Comment by xcompass** [comment]
Updated to 1.21.11. Issue fixed. Thanks.
